src/posthog-surveys.ts

```diff
@@ -99,6 +99,26 @@ export class PostHogSurveys {
                 const targetingFlagCheck = survey.targeting_flag_key
                     ? this.instance.featureFlags.isFeatureEnabled(survey.targeting_flag_key)
                     : true
+
+                let internalTargetingFlagCheck = undefined
+                if (survey.internal_targeting_flag_key) {
+                    // because this feature flag depends on user properties that might not be available yet,
+                    // the decide API will not return this flag when asked for it.
+                    // as a result, we need to distinguish between undefined and false.
+                    // where undefined means that the properties are not yet available.
+                    const value = this.instance.featureFlags.getFeatureFlag(survey.internal_targeting_flag_key)
```

Comment:
Arguably a nit, but can we pick a better name than `value` here?  I'd prefer something like `flagValue` or `featureFlagValue`.  Given that the rest of your variables are thoughtfully named, this one kinda surprised me.